language/bytecode-verifier/src/abstract_state.rs

```diff
@@ -173,45 +173,45 @@ impl AbstractState {
         self.locals
             .values()
             .all(|x| x.value.is_unrestricted_value())
-            && !self.is_nonce_borrowed(self.frame_root())
+            && !self.is_id_borrowed(self.frame_root())
     }
 
     /// destroys local@idx
     pub fn destroy_local(&mut self, idx: LocalIndex) {
         checked_precondition!(self.is_local_safe_to_destroy(idx));
         let local = self.locals.remove(&idx).unwrap();
         match local.value {
-            AbstractValue::Reference(nonce) => self.remove_nonce(nonce),
+            AbstractValue::Reference(id) => self.remove_id(id),
             AbstractValue::Value(kind) => {
                 checked_verify!(kind == Kind::Unrestricted);
             }
         }
     }
 
     /// Returns the frame root reference
-    fn frame_root(&self) -> Nonce {
-        Nonce::new(self.num_locals)
+    fn frame_root(&self) -> RefID {
+        RefID::new(self.num_locals)
     }
 
-    /// adds new nonce to borrow graph
-    pub fn add_nonce(&mut self) -> Nonce {
-        let nonce = Nonce::new(self.next_id);
-        self.borrow_graph.add_nonce(nonce);
+    /// adds new id to borrow graph
+    pub fn add_id(&mut self) -> RefID {
+        let id = RefID::new(self.next_id);
+        self.borrow_graph.add_id(id);
         self.next_id += 1;
-        nonce
+        id
     }
 
-    /// removes `nonce` from borrow graph
-    pub fn remove_nonce(&mut self, nonce: Nonce) {
-        self.borrow_graph.remove_nonce(nonce);
+    /// removes `id` from borrow graph
+    pub fn remove_id(&mut self, id: RefID) {
+        self.borrow_graph.remove_id(id);
     }
 
-    /// checks if `nonce` is borrowed
-    pub fn is_nonce_borrowed(&self, nonce: Nonce) -> bool {
-        !self.borrow_graph.all_borrows(nonce).is_empty()
+    /// checks if `id` is borrowed
+    pub fn is_id_borrowed(&self, id: RefID) -> bool {
```

Comment:
I would drop `id_` from the function name, `is_borrowed` sounds explicit enough to me... but as you wish... your code...